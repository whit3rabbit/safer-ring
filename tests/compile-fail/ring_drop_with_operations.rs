// This test should fail to compile because the ring cannot be dropped with operations in flight

use safer_ring::{Ring, PinnedBuffer};

fn main() {
    let buffer = PinnedBuffer::with_capacity(1024);
    
    {
        let ring = Ring::new(10).unwrap();
        let _future = ring.read(0, buffer.as_mut_slice());
        // Ring should not be droppable here with operation in flight
    } // This should cause a compile error
}