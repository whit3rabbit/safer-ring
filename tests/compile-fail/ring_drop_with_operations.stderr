error[E0597]: `ring` does not live long enough
  --> tests/compile-fail/ring_drop_with_operations.rs:9:22
   |
9  |         let _future = ring.read(0, buffer.as_mut_slice());
   |                       ^^^^ borrowed value does not live long enough
10 |         // Ring should not be droppable here with operation in flight
11 |     } // This should cause a compile error
   |     - `ring` dropped here while still borrowed
12 | }
   | - borrow might be used here, when `_future` is dropped and runs the `Drop` code for type `ReadFuture`